omod/src/main/java/org/openmrs/module/appui/UiSessionContext.java

```diff
@@ -82,6 +82,9 @@ public void setSessionLocation(Location sessionLocation) {
         }
         this.sessionLocation = sessionLocation;
         this.sessionLocationId = sessionLocation.getId();
+        if (userContext != null && userContext.getAuthenticatedUser() != null) {
```

Comment:
Does this also cater for the case where i just switch location when am already logged in?

Reply:
Yes, It will immediately swtich the location when you already logged in.